deploy/runtime/ui/Battery Dispatch PV Smoothing.txt

```diff
@@ -0,0 +1,1921 @@
+0
+0
+1
+1
+1
+1
+1.2
+Numeric
+batt_dispatch_pvs_max_ramp
+3
+1
+ Maximum ramp rate
+ % of nameplate per ramp interval
+Battery Dispatch PV Smoothing
+0
+0
+1
+1
+1
+1
+10
+Numeric
+batt_dispatch_pvs_nameplate
+3
+1
+Nameplate
+kWac
+Battery Dispatch PV Smoothing
+0
+8
+1
+1
+1
+1
+0
+Numeric
+batt_dispatch_pvs_nameplate_ac
+3
+1
+Power scaling factor
+kWac
+ 
+0
+8
+1
+1
+1
+1
+0
+Default
+batt_dispatch_pvs_ramp_interval
+3
+1
+ Ramp interval
+ minutes
+Battery Dispatch PV Smoothing
+0
+8
+1
+1
+1
+1
+10
+Numeric
+batt_dispatch_pvs_short_forecast_enable
+3
+1
+ 
+ 
+Battery Dispatch PV Smoothing
+0
+0
+1
+1
+1
+1
+0
+CheckBox
+batt_dispatch_pvs_soc_rest
+3
+1
+Battery resting SOC
+%
+Battery Dispatch PV Smoothing
+0
+0
+1
+1
+1
+1
+50
+Numeric
+batt_dispatch_pvs_timestep_multiplier
+3
+1
+Ramp timestep multiplier
+ 
+Battery Dispatch PV Smoothing
+0
+2
+1
+1
+1
+1
+3
+Numeric
+batt_dispatch_pvs_wf_timestep
+3
+1
+Weather file timestep
+minutes
+Battery Dispatch PV Smoothing
+0
+8
+1
+1
+1
+1
+60
+Numeric
+
+1484
+equations{'batt_dispatch_pvs_wf_timestep'} = define() {
+	return(${ui_step_minutes});
+};
+
+equations{'batt_dispatch_pvs_ramp_interval'} = define() {
+	return(${batt_dispatch_pvs_wf_timestep}*${batt_dispatch_pvs_timestep_multiplier});
+};
+
+equations{'batt_dispatch_pvs_battery_energy'} = define() {
+	return(${batt_computed_bank_capacity});
+};
+
+equations{'batt_dispatch_pvs_battery_power'} = define() {
+	return(${batt_power_charge_max_kwac});
+};
+
+equations{'batt_dispatch_pvs_interconnection_limit'} = define() {
+	return(${grid_interconnection_limit_kwac});
+};
+
+equations{'batt_dispatch_pvs_nameplate'} = define() {
+	if (technology() != "Generic Battery") {
+		return(${total_inverter_capacity});// todo - generic? PV array only?
+	}
+	else {
+		return ${system_capacity};
+	}
+};
+
+// todo - move to compute module?
+equations{'batt_dispatch_pvs_nameplate_ac'} = define() {
+	if (technology() != "Generic Battery") {
+		if (${total_inverter_capacity}>${grid_interconnection_limit_kwac}) 
+			return ${grid_interconnection_limit_kwac};
+		else
+			return ${total_inverter_capacity};
+	}
+	else {
+		if (${system_capacity}>${grid_interconnection_limit_kwac})
+			return ${grid_interconnection_limit_kwac};
+		else
+			return ${system_capacity};
+		}
+};
+
+
+equations{'batt_dispatch_pvs_battery_rte'} = define() {
+		if (${batt_ac_or_dc} == 1) {
+			return (${batt_ac_dc_efficiency}*${batt_dc_ac_efficiency}/100.0);
+		}
+		else {
+			return (${batt_dc_dc_efficiency}*${batt_inverter_efficiency_cutoff}/100.0);
+		}
+	};
+
+2523
+on_load{'Battery Dispatch PV Smoothing'} = define() {
+	on_change{'batt_dispatch_pvs_timestep_multiplier'}();
+	on_change{'batt_dispatch_pvs_ac_ub_enable'}();
+	on_change{'batt_dispatch_pvs_ac_lb_enable'}();
+	on_change{'batt_dispatch_pvs_short_forecast_enable'}();
+	value('batt_dispatch_choice', 5);
+	toggle_dispatch_pvs(); // do not enable unless mode == 5
+};
+
+function toggle_dispatch_pvs()
```

Comment:
Change toggle_dispatch to toggle_dispatch_pvs as Manual Dispatch also uses toggle_dispatch()